frontend/lib/editor/initiative.ts

```diff
@@ -97,78 +80,16 @@ const createEditorInitiative = (initiative: Initiative): EditorInitiative => {
   }
 }
 
-const computedInitiative = (ref: Ref<EditorInitiative>): ComputedRef<Initiative> => {
-  const v = ref.value
-  return computed(() => ({
-    ...asValue(v.id),
-    ...asValue(v.name),
-    ...asValue(v.affiliation),
-    ...asValue(v.publicDescription),
-    ...asValue(v.internalDescription),
-    ...asValue(v.requiresInvitationToJoin),
-    ...asValue(v.isAcceptingNewMembers),
-    ...asValue(v.isAcceptingNewPortfolios),
-    ...asValue(v.language),
-    ...asValue(v.pactaVersion),
-    ...asValue(v.createdAt),
-  }))
-}
-
-const getComputedInitiativeChanges = (ref: Ref<EditorInitiative>): ComputedRef<InitiativeChanges> => {
-  return computed(() => {
-    const v = ref.value
-    return {
-      ...asChange(v.id),
-      ...asChange(v.name),
-      ...asChange(v.affiliation),
-      ...asChange(v.publicDescription),
-      ...asChange(v.internalDescription),
-      ...asChange(v.requiresInvitationToJoin),
-      ...asChange(v.isAcceptingNewMembers),
-      ...asChange(v.isAcceptingNewPortfolios),
-      ...asChange(v.language),
-      ...asChange(v.pactaVersion),
-      ...asChange(v.createdAt),
-    }
-  })
+export const initiativeEditor = (i: Initiative): EditorComputedValues<Initiative> => {
+  return getEditorComputedValues(
+    'lib/editor/initiative', i, createEditorInitiative)
 }
 
-const getComputedIncompleteFields = (ref: Ref<EditorInitiative>): ComputedRef<string[]> => {
-  return computed(() => {
-    const v = ref.value
-    return [
-      ...asIncompleteField(v.id),
-      ...asIncompleteField(v.name),
-      ...asIncompleteField(v.affiliation),
-      ...asIncompleteField(v.publicDescription),
-      ...asIncompleteField(v.internalDescription),
-      ...asIncompleteField(v.requiresInvitationToJoin),
-      ...asIncompleteField(v.isAcceptingNewMembers),
-      ...asIncompleteField(v.isAcceptingNewPortfolios),
-      ...asIncompleteField(v.language),
-      ...asIncompleteField(v.pactaVersion),
-      ...asIncompleteField(v.createdAt),
-    ]
-  })
+interface Example {
+  a: string
+  b: string | undefined
+  c?: string
+  d?: string | undefined
 }
 
-export const initiativeEditor = (i: Initiative) => {
-  const ei = useState<EditorInitiative>('lib/editor/initiative')
-  ei.value = createEditorInitiative(i)
-  const incompleteFields = getComputedIncompleteFields(ei)
-  const changes = getComputedInitiativeChanges(ei)
-  const initiative = computedInitiative(ei)
-  const setInitiative = (pv: Initiative) => { ei.value = createEditorInitiative(pv) }
-  const hasChanges = computed(() => changes.value && Object.keys(changes.value).length > 0)
-  const isIncomplete = computed(() => incompleteFields.value.length > 0)
-
-  return {
-    editorInitiative: ei,
-    setInitiative,
-    incompleteFields,
-    changes,
-    hasChanges,
-    isIncomplete,
-    initiative,
-  }
-}
+export type ExampleEditor = EditorFieldsFor<Example>
```

Comment:
What's the purpose of this?

Reply:
GQ - it's used as a prop input type to `InitiativeEditor.vue`.

Reply:
Oh nvm. I see what you mean. Removed. I was playing around with this to understand how ` | undefined` differed from `var?:`